sdks/go/pkg/beam/core/runtime/graphx/tree.go

```diff
@@ -0,0 +1,121 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graphx
+
+import (
+	"fmt"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/graph"
+)
+
+// NamedEdge is a named MultiEdge.
+type NamedEdge struct {
+	Name string
+	Edge *graph.MultiEdge
+}
+
+// NamedScope is a named Scope.
+type NamedScope struct {
+	Name  string
+	Scope *graph.Scope
+}
+
+// ScopeTree is a convenient representation of the Scope-structure as a tree.
+// Each ScopeTree may also be a subtree of another ScopeTree. The tree structure
+// must respect the underlying Scope structure, i.e., if Scope 'a' has a parent
```

Comment:
The tree structure respects the underlying...

Reply:
Removed "must".